### PR TITLE
add helper functions

### PR DESCRIPTION
**Pull request by klizhentas**

### CODE REVIEW

File: lib.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import (
+	"reflect"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// GetStringMapValue is a helper function that returns property
+// from map[string]string or map[string][]string
+// GetMapValure returns empty value in case if key not found
```
Comment (r0mant) [inline lib.go]: GetMapValure -> GetStringMapValue?

File: lib.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+// In case if map is nil, returns empty value as well
+func GetStringMapValue(mapVal, keyVal interface{}) (interface{}, error) {
+	key, ok := keyVal.(string)
+	if !ok {
+		return nil, trace.BadParameter("")
```
Comment (r0mant) [inline lib.go]: Add some text to BadParameter message?

File: lib.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	}
+	switch m := mapVal.(type) {
+	case map[string][]string:
+		if len(m) == 0 {
+			var n []string
```
Comment (r0mant) [inline lib.go]: You could just do `return []string{}, nil` so you don't need the var?

Comment (klizhentas) [reply lib.go]: to return nil of a proper type

**Pull request merged**